### PR TITLE
Simulated "false start" fix + default race change

### PR DESCRIPTION
**Pull request by simonttp78**
1. Fix (in fact restore from Arduino IDE version) for simulated "false start". Without that change simulated race with false starts of first dog doesn't work properly.
2. Also default race changed to Race 3 with nice demo/scenario that include "false start" and rerun of first dog.
3. some formatting / cosmetic changes made by "Prettier - Code formatter" extension of VSCode

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -22,6 +22,11 @@ board = lolin32
 framework = arduino
 lib_ldf_mode = deep+
 monitor_speed = 115200
+upload_protocol = espota
+upload_port = 192.168.20.1
+upload_flags =
+    --host_ip=192.168.20.2
```
Comment (vyruz1986) [inline platformio.ini]: This is not guaranteed to be the host IP, e.g. if you connect with 2 devices, the 2nd device will get `192.168.20.3` as the IP address. Are you sure this parameter is required?

File: lib/Simulator/Simulator.cpp [hunk 2]
```diff
@@ -83,18 +112,15 @@ void SimulatorClass::Main()
       //Pending record doesn't contain valid data, this means we've reched the end of our queue
       return;
    }
+   long lRaceElapsedTime = GET_MICROS - RaceHandler._lRaceStartTime;
    //Simulate sensors
-   if (RaceHandler.RaceState != RaceHandler.STOPPED
-      && PendingRecord.lTriggerTime <= (long) RaceHandler._lRaceTime)
+   if (RaceHandler.RaceState != RaceHandler.STOPPED && PendingRecord.lTriggerTime <= (long)lRaceElapsedTime)
```
Comment (vyruz1986) [inline lib/Simulator/Simulator.cpp:117]: Why did you change this? `RaceHandler._lRaceTime` is continuously calculated in `RaceHandler.Main()`, so you're doing the same calculation here again (on line 115), but if you use `RaceHandler._lRaceTime` you get the same number.

File: lib/Simulator/Simulator.cpp [hunk 3]
```diff
@@ -83,18 +112,15 @@ void SimulatorClass::Main()
...
    {
       if (RaceHandler._QueueEmpty())
       {
-         if ((PendingRecord.lTriggerTime < 0
-            && RaceHandler.RaceState == RaceHandler.STARTING)
-            || (PendingRecord.lTriggerTime > 0
-            && RaceHandler.RaceState == RaceHandler.RUNNING))
+         if ((PendingRecord.lTriggerTime < 0 && RaceHandler.RaceState == RaceHandler.STARTING) || (PendingRecord.lTriggerTime > 0 && RaceHandler.RaceState == RaceHandler.RUNNING))
```
Comment (vyruz1986) [inline lib/Simulator/Simulator.cpp]: Can you please change this back to multiple lines? If all the comparison statements are on the same line, it is not possible to see everything in 1 screen when viewing the code in an editor.

**Review by vyruz1986 — CHANGES_REQUESTED** [review]
Hi again Simon!

Thanks for the pull request! As far as the default simulator records, I don't care much which one is in there, so it's for me to change it.

Regarding the other changes you did, I have some remarks, please see my comments in line and either change them or provide some feedback.

File: platformio.ini (hunk 1 above)
Comment (simonttp78) [reply platformio.ini]: You are right. I had issues due to working on Virtual Machine, but it should not be mandatory. I change it to comment as it might be helpful for someone.

File: lib/Simulator/Simulator.cpp (hunk 2 above)
Comment (simonttp78) [reply lib/Simulator/Simulator.cpp:117]: Without this change (that was in original release for Arduino) simulator doesn't work for cases/races with "false start" as it subtract negative triggers (e.g. -102316) from "race start time" that is not yet updated and is equal to time of pressing START button still. Effect is that fault instead be seen 102ms before green light, it's indicated 102ms after START sequence is triggered. I hope you got the point.

File: lib/Simulator/Simulator.cpp (hunk 3 above)
Comment (simonttp78) [reply lib/Simulator/Simulator.cpp]: That is kind of fault in "prettier" extension of VSCode. While typically it wrap lines wider that 80 character for better visibility somehow in "if" it do exactly opposite :-(
I changed to 2 lines that I think is good solution from visibility p.o.v.. Please check and let me know. I can revert to original 4 lines if you do prefer that.

**Comment by simonttp78** [comment]
Hi Alex,

I did updates in pull request. Please review them.
In addition I made 10min clip on YouTube. Please have a look and comment:
https://youtu.be/cdUDgUvf9Vc <https://youtu.be/cdUDgUvf9Vc>

Best regards,
Simon

File: lib/Simulator/Simulator.cpp (hunk 2 above)
Comment (vyruz1986) [reply lib/Simulator/Simulator.cpp:117]: Clear, thanks!

**Review by vyruz1986 — APPROVED** [review]

**Comment by vyruz1986** [comment]
Looks all good now. Thanks for the video, I left a comment there. I will also work on an `.editorconfig` file and some recommended plugins for VSCode so we can minimize the white space differences as much as possible.
